include/flecs/flecs.hpp

```diff
@@ -0,0 +1,1571 @@
+#pragma once
+
+#include <string>
+#include <sstream>
+#include <array>
+
+namespace flecs {
+
+////////////////////////////////////////////////////////////////////////////////
+//// Forward declarations and types
+////////////////////////////////////////////////////////////////////////////////
+
+using world_t = ecs_world_t;
+using entity_t = ecs_entity_t;
+using type_t = ecs_type_t;
+using snapshot_t = ecs_snapshot_t;
+using filter_t = ecs_filter_t;
+using filter_iter_t = ecs_filter_iter_t;
+
+class world;
+class snapshot;
+class entity;
+class type;
+class rows;
+class filter;
+class filter_iterator;
+class world_filter;
+class snapshot_filter;
+
+template <typename T>
+class component_base;
+
+enum system_kind {
+    OnLoad = EcsOnLoad,
+    PostLoad = EcsPostLoad,
+    PreUpdate = EcsPreUpdate,
+    OnUpdate = EcsOnUpdate,
+    OnValidate = EcsOnValidate,
+    PostUpdate = EcsPostUpdate,
+    PreStore = EcsPreStore,
+    OnStore = EcsOnStore,
+    Manual = EcsManual,
+    OnAdd = EcsOnAdd,
+    OnRemove = EcsOnRemove,
+    OnSet = EcsOnSet
+};
+
+enum match_kind {
+    MatchDefault = EcsMatchDefault,
+    MatchAll = EcsMatchAll,
+    MatchAny = EcsMatchAny,
+    MatchExact = EcsMatchExact
+};
+
+
+////////////////////////////////////////////////////////////////////////////////
+//// Builtin components and tags 
+////////////////////////////////////////////////////////////////////////////////
+
+using Component = EcsComponent;
+using TypeComponent = EcsTypeComponent;
+using Prefab = EcsPrefab;
+static const entity_t PrefabParent = EEcsPrefabParent;
+static const entity_t PrefabBuilder = EEcsPrefabBuilder;
+static const entity_t RowSystem = EEcsRowSystem;
+static const entity_t ColSystem = EEcsColSystem;
+using Name = EcsId;
+static const entity_t Hidden = EEcsHidden;
+static const entity_t Disabled = EEcsDisabled;
+static const entity_t OnDemand = EEcsOnDemand;
+
+static const entity_t World = EcsWorld;
+static const entity_t Singleton = EcsSingleton;
+static const entity_t Invalid = EcsInvalid;
+
+
+////////////////////////////////////////////////////////////////////////////////
+//// Wrapper class around a table column
+////////////////////////////////////////////////////////////////////////////////
+
+template <typename T>
+class column final {
```

Comment:
Why the column is marked as final, without inheritance ?

Reply:
I prevent inheriting from framework classes wherever possible, just because it's bad practice.